File: scripts/logout/logout.php
<?php
    /* change 'central' to the url of your weblogin server */
    $central = "https://weblogin.umich.edu/cgi-bin/logout";

    setcookie( $_SERVER[ 'COSIGN_SERVICE' ], "null", time()-1, '/', "", 1 );

    /* make any local additions here (e.g. expiring local sessions, etc.),
       but it's important that there be no output on this page. */

    header( "Location: $central" );
    exit;
?>

File: scripts/logout/logout.pl
#!/usr/local/bin/perl -wT

use strict;

# change 'central' to the url of your weblogin server.
my $central = "https://weblogin.umich.edu/cgi-bin/logout";
my $query_string = "";

# expire and nullify service cookie
print( "Set-Cookie: $ENV{ COSIGN_SERVICE }=null; path=/; expires=Wednesday, 27-Jan-77 00:00:00 GMT; secure\n" );

if ( $ENV{ QUERY_STRING } =~ m|^(https?://.*)$| ) {
    $query_string = "?$1";
}

# perform any local cleanup here

# redirect to central weblogin server
print( "Location: $central$query_string\n\n" );

exit( 0 );
